### PR TITLE
feat(installer): replace shell interpreter with bash, add version checks

### PR DESCRIPTION
**Pull request by gfranxman**
for dependencies

- The shell interpreter from sh has been changed to bash. 
- The install script now also includes a version checking mechanism for dependencies. 
    - This ensures that the required versions of dependencies are installed. 
    - The install script also includes comments detailing its purpose and usage. 
- The shell command in the GitHub workflow and README installation instructions have been updated accordingly.
- Debug log messages have been added for troubleshooting purposes.

### CODE REVIEW

File: install.sh [hunk 1]
```diff
@@ -32,68 +62,106 @@ command_exists() {
...
+        case $tool_name in
             "bash")
                 echo "Please install bash for consistent shell execution."
                 ;;
             "curl")
                 echo "Please install curl:"
-                echo "  - Ubuntu/Debian: sudo apt-get install curl"
-                echo "  - CentOS/Fedora: sudo yum install curl"
-                echo "  - macOS: brew install curl"
+                echo "  - Ubuntu/Debian: sudo apt-get update && sudo apt-get install --only-upgrade curl"
```
Comment (rochecompaan) [inline install.sh:140]:
```suggestion
                echo "  - Arch: sudo pacman -Sy --needed curl"
                echo "  - Ubuntu/Debian: sudo apt-get update && sudo apt-get install --only-upgrade curl"
```

File: install.sh [hunk 2]
```diff
@@ -32,68 +62,106 @@ command_exists() {
...
+                echo "  - CentOS/Fedora: sudo yum install curl || sudo yum upgrade curl"
+                echo "  - macOS: brew install curl || brew upgrade curl"
                 echo "  - Windows: Download from https://curl.se/windows/"
                 ;;
             "make")
                 echo "Please install make:"
-                echo "  - Ubuntu/Debian: sudo apt-get install make"
-                echo "  - CentOS/Fedora: sudo yum install make"
+                echo "  - Ubuntu/Debian: sudo apt-get update && sudo apt-get install --only-upgrade make"
```
Comment (rochecompaan) [inline install.sh:147]:
```suggestion
                echo "  - Arch: sudo pacman -Sy --needed make"
                echo "  - Ubuntu/Debian: sudo apt-get update && sudo apt-get install --only-upgrade make"
```

File: install.sh [hunk 3]
```diff
@@ -32,68 +62,106 @@ command_exists() {
...
                 echo "  - macOS: xcode-select --install"
                 echo "    or: brew install make"
                 echo "  - Windows: Install MinGW or use WSL"
                 ;;
             "python3")
                 echo "Please install python3:"
-                echo "  - Ubuntu/Debian: sudo apt-get install python3"
-                echo "  - CentOS/Fedora: sudo yum install python3"
-                echo "  - macOS: brew install python"
+                echo "  - Ubuntu/Debian: sudo apt-get update && sudo apt-get install --only-upgrade python3"
```
Comment (rochecompaan) [inline install.sh:155]:
```suggestion
                echo "  - Arch: sudo pacman -Sy --needed python3"
                echo "  - Ubuntu/Debian: sudo apt-get update && sudo apt-get install --only-upgrade python3"
```

File: install.sh [hunk 4]
```diff
@@ -32,68 +62,106 @@ command_exists() {
...
+                echo "  - CentOS/Fedora: sudo yum install python3 || sudo yum upgrade python3"
+                echo "  - macOS: brew install python || brew upgrade python"
                 echo "  - Windows: Download from https://www.python.org/downloads/windows/"
                 ;;
             "rsync")
                 echo "Please install rsync:"
-                echo "  - Ubuntu/Debian: sudo apt-get install rsync"
-                echo "  - CentOS/Fedora: sudo yum install rsync"
-                echo "  - macOS: brew install rsync"
+                echo "  - Ubuntu/Debian: sudo apt-get update && sudo apt-get install --only-upgrade rsync"
```
Comment (rochecompaan) [inline install.sh:162]:
```suggestion
                echo "  - Arch: sudo pacman -Sy --needed rsync"
                echo "  - Ubuntu/Debian: sudo apt-get update && sudo apt-get install --only-upgrade rsync"
```

File: install.sh [hunk 5]
```diff
@@ -32,68 +62,106 @@ command_exists() {
...
+                echo "  - CentOS/Fedora: sudo yum install rsync || sudo yum upgrade rsync"
+                echo "  - macOS: brew install rsync || brew upgrade rsync"
                 echo "  - Windows: Can be obtained from https://www.itefix.net/cwrsync"
                 echo "    or: Install WSL and usesudo apt-get install rsync"
```
Comment (rochecompaan) [inline install.sh:166]:
```suggestion
                echo "    or: Install WSL and use sudo apt-get install rsync"
```

**Review by rochecompaan — APPROVED** [review]
Nice! I added a few suggestions to install/upgrade dependencies on Arch.

**Comment by clshaw01** [comment]
not needed

**Pull request closed without merging**